app/models/metric/ci_mixin/capture.rb

```diff
@@ -177,6 +177,8 @@ def perf_capture(interval_name, start_time = nil, end_time = nil)
 
     if start_range.nil?
       _log.info "#{log_header} Skipping processing for #{log_target} as no metrics were captured."
+      # Set the last capture on to end_time to prevent forever queueing up the same collection range
+      update_attribute(:last_perf_capture_on, end_time) if last_perf_capture_on.nil?
```

Comment:
Looks good, although I think we want `update` or `update_attributes` so it does validations I believe